cpufreq/src/cpufreq-prefs.c

```diff
@@ -31,554 +31,564 @@
+        gtk_widget_set_sensitive (prefs->priv->show_perc,
+                                  (TRUE && key_writable));
+
+        if (prefs->priv->show_text_mode == CPUFREQ_MODE_TEXT_PERCENTAGE)
+            gtk_widget_set_sensitive (prefs->priv->show_unit,
+                                      FALSE);
+        else
+            gtk_widget_set_sensitive (prefs->priv->show_unit,
+                                      (TRUE && key_writable));
+    } else {
+        gtk_widget_set_sensitive (prefs->priv->show_freq, FALSE);
+        gtk_widget_set_sensitive (prefs->priv->show_unit, FALSE);
+        gtk_widget_set_sensitive (prefs->priv->show_perc, FALSE);
+    }
 }
 
 static void
 cpufreq_prefs_dialog_update (CPUFreqPrefs *prefs)
 {
-	if (cpufreq_utils_get_n_cpus () > 1) {
-		gtk_combo_box_set_active (GTK_COMBO_BOX (prefs->priv->cpu_combo),
-					  MIN (prefs->priv->cpu, cpufreq_utils_get_n_cpus () - 1));
-	}
-
-	gtk_combo_box_set_active (GTK_COMBO_BOX (prefs->priv->show_mode_combo),
-				  prefs->priv->show_mode);
-	
-	switch (prefs->priv->show_text_mode) {
-	case CPUFREQ_MODE_TEXT_FREQUENCY:
-		gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (prefs->priv->show_freq),
-					      TRUE);
-		gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (prefs->priv->show_unit),
-					      FALSE);
-
-		break;
-	case CPUFREQ_MODE_TEXT_FREQUENCY_UNIT:
-		gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (prefs->priv->show_freq),
-					      TRUE);
-		gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (prefs->priv->show_unit),
-					      TRUE);
-
-		break;
-	case CPUFREQ_MODE_TEXT_PERCENTAGE:
-		gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (prefs->priv->show_perc),
-					      TRUE);
-
-		break;
-	}
+    if (cpufreq_utils_get_n_cpus () > 1) {
+        gtk_combo_box_set_active (GTK_COMBO_BOX (prefs->priv->cpu_combo),
+                                  MIN (prefs->priv->cpu,
+                                  cpufreq_utils_get_n_cpus () - 1));
+    }
+
+    gtk_combo_box_set_active (GTK_COMBO_BOX (prefs->priv->show_mode_combo),
+                              prefs->priv->show_mode);
+
+    switch (prefs->priv->show_text_mode) {
+    case CPUFREQ_MODE_TEXT_FREQUENCY:
+        gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (prefs->priv->show_freq),
+                                      TRUE);
+        gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (prefs->priv->show_unit),
+                                      FALSE);
+
+        break;
+    case CPUFREQ_MODE_TEXT_FREQUENCY_UNIT:
+        gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (prefs->priv->show_freq),
+                                      TRUE);
+        gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (prefs->priv->show_unit),
+                                      TRUE);
+
+        break;
+    case CPUFREQ_MODE_TEXT_PERCENTAGE:
+        gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (prefs->priv->show_perc),
+                                      TRUE);
+
+        break;
+    }
 }
```

Comment:
@rbuj 
Is a `default` missing inthe case scope?

Reply:
```C
/* gcc -Wall switch.c -o switch */
#include <stdio.h>

void test (char option)
{
   if (option == 'A') {
        printf ("<%c>", option);
	goto end;
   }

   switch (option) {
        case 'a':
        case 'b':
                printf ("a or b?, %c\n", option);
        case 'c':
                printf ("a, b or c? %c", option);
		break;
        case 'd': {
		int i;
		for (i=0;i<9;i++)
	                printf ("%d", i);
		printf (" %c", option);
                break;
        }
	case 'e':
		printf ("%c", option);
        default:
               printf ("?");
               /* goto end; */
   }
end:
   printf ("\n---\n");
}

int main (void)
{
   test ('a');
   test ('b');
   test ('c');
   test ('d');
   test ('e');
   test ('z');
   test ('A');

   return 0;
}
```
```
$ ./switch 
a or b?, a
a, b or c? a
---
a or b?, b
a, b or c? b
---
a, b or c? c
---
012345678 d
---
e?
---
?
---
<A>
---
```

Reply:
You can use `g_assert_not_reached ()` in order to check unexpected value of `prefs->priv->show_text_mode`:
```C
default:
    g_assert_not_reached ();
```